cloud/aws/alb/monitors-alb.tf

```diff
@@ -0,0 +1,203 @@
+		detect(when(signal > ${var.httpcode_5xx_threshold_warning}) and when(B > ${var.httpcode_5xx_threshold_number_requests})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "are too high > ${var.httpcode_5xx_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.httpcode_5xx_disabled_critical, var.httpcode_5xx_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.httpcode_5xx_notifications_critical, var.httpcode_5xx_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "are too high > ${var.httpcode_5xx_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.httpcode_5xx_disabled_warning, var.httpcode_5xx_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.httpcode_5xx_notifications_warning, var.httpcode_5xx_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "httpcode_4xx" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] ALB 4xx errors"
+
+	program_text = <<-EOF
+		A = data('HTTPCode_ELB_4XX_Count', filter=filter('namespace', 'AWS/ApplicationELB') and filter('stat', 'sum') and (not filter('AvailabilityZone', '*')) and ${module.filter-tags.filter_custom}, extrapolation='zero'){var.httpcode_4xx_aggregation_function}
+		B = data('RequestCount', filter=filter('namespace', 'AWS/ApplicationELB') and filter('stat', 'sum') and ${module.filter-tags.filter_custom}, extrapolation='zero'){var.httpcode_4xx_aggregation_function}
+		signal = (A/B).scale(100).${var.httpcode_4xx_transformation_function}(over='${var.httpcode_4xx_transformation_window}')
+		detect(when(signal > ${var.httpcode_4xx_threshold_critical}) and when(B > ${var.httpcode_4xx_threshold_number_requests})).publish('CRIT')
+		detect(when(signal > ${var.httpcode_4xx_threshold_warning}) and when(B > ${var.httpcode_4xx_threshold_number_requests})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "are too high > ${var.httpcode_4xx_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.httpcode_4xx_disabled_critical, var.httpcode_4xx_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.httpcode_4xx_notifications_critical, var.httpcode_4xx_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "are too high > ${var.httpcode_4xx_threshold_warning}"
```

Comment:
```suggestion
 		description           = "are too high > ${var.httpcode_4xx_threshold_warning}%"
```

Reply:
Proper english would be 'are' not 'is' because errors are plural.  Rule is "it is or they are".  I.E.  The rate is too high, but 4xx errors are too high